src/common/librlist/rlist.c

```diff
@@ -890,16 +1043,19 @@ struct rlist *rlist_from_json (json_t *o, json_error_t *errp)
     json_t *R_lite = NULL;
     json_t *nodelist = NULL;
     json_t *scheduling = NULL;
+    json_t *properties = NULL;
     double starttime = -1.;
     double expiration = -1.;
+    flux_error_t error;
 
     if (json_unpack_ex (o, errp, 0,
-                        "{s:i s?O s:{s:o s?o s?F s?F}}",
+                        "{s:i s?O s:{s:o s?o s?o s?F s?F}}",
```

Comment:
`nodelist` is mandatory and must exist?

Reply:
yes, from [RFC 20](https://flux-framework.readthedocs.io/projects/flux-rfc/en/latest/spec_20.html)

> The value of the execution key SHALL contain at least the keys R_lite, and nodelist, with optional keys properties, starttime and expiration. Other keys are reserved for future extensions.

I don't think the `nodelist` requirement changed in this PR?

Reply:
Yes, this PR doesn't change `nodelist` requirement so it is okay not to change. I just noticed the the unpack was treating 'nodelist` as optional.

Reply:
Ah! ok I see. It might be that works out well for testing, but we can look at improving it to required in the future.